tests/unit/modules/test_gpg.py

```diff
@@ -378,9 +379,410 @@ def test_delete_key(self):
+                        ret = gpg.delete_key(
+                            "xxxxxxxxxxxxxxxx", delete_secret=True, use_passphrase=False
+                        )
+                        self.assertEqual(ret, _expected_result)
+                        gnupg_delete_keys.assert_called_with(
+                            "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+                            False,
+                            expect_passphrase=False,
+                        )
+
+    @pytest.mark.destructive_test  # Need to run as root!?
+    @skipIf(not HAS_GPG, "GPG Module Unavailable")
+    def test_delete_key_with_passphrase_without_gpg_passphrase_in_pillar(self):
+        """
+        Test gpg.delete_key with passphrase but without gpg_passphrase pillar
+        """
+
+        _user_mock = {
+            "shell": "/bin/bash",
+            "workphone": "",
+            "uid": 0,
+            "passwd": "x",
+            "roomnumber": "",
+            "gid": 0,
+            "groups": ["root"],
+            "home": self.gpghome,
+            "fullname": "root",
+            "homephone": "",
+            "name": "root",
+        }
+
+        _list_result = [
+            {
+                "dummy": "",
+                "keyid": "xxxxxxxxxxxxxxxx",
+                "expires": "2011188692",
+                "sigs": [],
+                "subkeys": [
+                    [
+                        "xxxxxxxxxxxxxxxx",
+                        "e",
+                        "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+                    ]
+                ],
+                "length": "4096",
+                "ownertrust": "-",
+                "sig": "",
+                "algo": "1",
+                "fingerprint": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+                "date": "1506612692",
+                "trust": "-",
+                "type": "pub",
+                "uids": ["GPG Person <person@example.com>"],
+            }
+        ]
+
+        _expected_result = {
+            "res": True,
+            "message": "gpg_passphrase not available in pillar.",
+        }
+
+        mock_opt = MagicMock(return_value="root")
+        pillar_mock = MagicMock(return_value=None)
+        with patch.dict(
+            gpg.__salt__, {"user.info": MagicMock(return_value=_user_mock)}
+        ):
+            with patch.dict(gpg.__salt__, {"config.option": mock_opt}), patch.dict(
+                gpg.__salt__, {"pillar.get": pillar_mock}
+            ):
+                with patch.object(gpg, "_list_keys", return_value=_list_result):
+                    with patch(
+                        "salt.modules.gpg.gnupg.GPG.delete_keys"
+                    ) as gnupg_delete_keys:
                         ret = gpg.delete_key("xxxxxxxxxxxxxxxx", delete_secret=True)
                         self.assertEqual(ret, _expected_result)
+                        gnupg_delete_keys.assert_not_called()
+
+    @pytest.mark.destructive_test  # Need to run as root!?
```

Comment:
And why is it destructive?
If it messes with tempfiles, and not the actual user's gpg keys, it's not destructive.

And if we're mocking, it's really not destructive, right?

Reply:
The same of the previous one.